tests/datasets/test_marketdata.py

```
@@ -0,0 +1,116 @@
+
+    def test_download_without_credentials(self):
+        """Test the download method handles missing Kaggle credentials."""
+        # Temporarily rename the Kaggle config file if it exists
+        kaggle_config = os.path.expanduser("~/.kaggle/kaggle.json")
+        if os.path.exists(kaggle_config):
+            temp_config = kaggle_config + ".temp"
+            os.rename(kaggle_config, temp_config)
+        else:
+            temp_config = None
+        try:
+            with self.assertRaises(Exception):
```

Ruff (0.8.0)

`assertRaises(Exception)` should be considered evil

(B017)